src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import com.nixxcode.jvmbrotli.common.BrotliLoader;
+import com.nixxcode.jvmbrotli.enc.BrotliOutputStream;
+import com.nixxcode.jvmbrotli.enc.Encoder;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class BrotliEncoder extends MessageToByteEncoder<ByteBuf> {
+
+  private static final Logger log = LoggerFactory.getLogger(BrotliEncoder.class);
+
+  private final boolean preferDirect;
+  private final int compressionQuality;
+  private final int windowSize;
+
+  /*
+   If the Brotli encoding is being used to compress streams in real-time,
+   it is not advisable to have a quality setting above 4 due to performance.
+  */
+  private static final int DEFAULT_COMPRESSION_QUALITY = 4;
+
+  /*
+   If the Brotli encoding is being used to compress streams in real-time,
```

Comment:
comment seems to be identical to the above, can we describe this one properly?

Reply:
fixed